core/src/test/java/io/grpc/internal/KeepAliveManagerTest.java

```diff
@@ -71,8 +73,8 @@ public long read() {
   @Before
   public void setUp() {
     MockitoAnnotations.initMocks(this);
-    keepAliveManager = new KeepAliveManager(transport, scheduler, ticker, 1000, 2000, false);
-    keepAliveManager.onTransportStarted();
+    keepAlivePinger = new ClientKeepAlivePinger(transport);
```

Comment:
Could you use a mock for the pinger for the majority of the tests? Then you can have a few tests to test the ClientKeepAlivePinger by itself, and those could mock the transport.